[Ron Quah] iP

salads-source opened the pull request:
# Implementation Progress for iP :rocket:

## Key Features Added
- Add support for multiple task types: **ToDo, Deadline, Event**
- Implement error handling for invalid commands
- Include a clean CLI-based user interface 😉

### Benefits of This PR
1. Improves code maintainability by adhering to OOP principles
2. Enhances user experience with a more detailed task manager
3. Increases test coverage and ensures fewer runtime issues

### Entry Point To This Application
```java
public static void main(String[] args) {
        (new Ron("./data/ron.txt")).run();
}
```
### Application Download
- Feel free to download a copy of the application [here](https://github.com/salads-source/ip/releases/tag/A-Jar)
- Once downloaded, run this command to start the application: `java -jar ron.jar`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

salads-source committed:
refactor: change chatbot name

salads-source committed:
Add: Greet and exit message

salads-source committed:
Add: Echo commands and exits when user enters bye

salads-source committed:
Add: Add and list commands

salads-source committed:
Add: Task class

salads-source committed:
Add: Ability to mark task as done or undone

salads-source committed:
Refactor: Main method to handle exceptions

salads-source committed:
Add: Subclasses for different tasks

salads-source committed:
Add: event handling for specific tasks

salads-source committed:
Test: Unit testing for main class

salads-source committed:
Add: Exception handling for invalid inputs

salads-source committed:
Add: Delete tasks functionality

salads-source committed:
Refactor: Logic from main method to separate static methods

salads-source committed:
Refactor: Remove redundant line separators

salads-source committed:
Complete Level 7 implementation

salads-source committed:
Merge branch-Level-7 to master

salads-source committed:
Complete Level-8 implementation

salads-source committed:
Merge branch-Level-8 to master

salads-source committed:
Add Ui class

salads-source committed:
Add TaskList class

salads-source committed:
Add Parser class

salads-source committed:
Refactor main class

salads-source committed:
Merge A-MoreOOP branch to master

salads-source committed:
Divide classes into packages

salads-source committed:
Merge A-Packages branch to master

salads-source committed:
Merge A-Gradle branch to master

salads-source committed:
Edit application entry point

salads-source committed:
Add JUnit testing for Parser and Storage class

salads-source committed:
Merge A-JUnit branch to master

salads-source committed:
Change gradle configuration to create jar file

salads-source committed:
Merge A-Jar branch to master

salads-source committed:
Add JavaDocs for classes and methods

salads-source committed:
Fix code structure to comply with Java coding standard

salads-source committed:
Implement Level-9

salads-source committed:
Merge branch-A-JavaDoc branch to master

salads-source committed:
Resolve merge conflict between master and branch-A-CodingStandard

salads-source committed:
Resolve merge conflict between master and branch-Level-9

salads-source committed:
Include this() keyword for better readability

salads-source committed:
Add JavaDocs for new methods

salads-source committed:
Refactor commands into a parent abstract Command class and child subcommand classes

salads-source committed:
Fix import statements to list imported classes explicitly

salads-source committed:
Add checkStyle configuration to detect coding style violations

salads-source committed:
Merge branch-A-CheckStyle branch to master branch

salads-source committed:
Add base configuration for javafx

salads-source committed:
Add resources for javafx

salads-source committed:
Refactor code to run application as javafx gui

salads-source committed:
Merge branch-Level-10 branch to master branch

salads-source committed:
Add javadocs for Level-10

salads-source committed:
Merge branch-Level-10 branch to master branch

juniper285 commented on src/main/java/ron/task/Task.java:
```diff
@@ -0,0 +1,89 @@
+ */
+public abstract class Task {
+    private final String name;
+    private boolean completed = false;
```
i'm just nitpicking, but perhaps you can consider using isCompleted for all your Tasks, instead of just completed?

> Boolean variables/methods should be named to sound like booleans
> As much as possible, use a prefix such as is, has, was, etc. for boolean variable/method names so that linters can automatically verify that this style rule is being followed.

juniper285 reviewed (commented):
LGTM, it looks very clean, only very very small details that does not affect anything. i found it funny that your bot is named ron too LOL

Casielim commented on src/main/java/ron/command/MarkCommand.java:
```diff
@@ -0,0 +1,23 @@
+ * Mark stored commands
+ */
+public class MarkCommand extends Command {
+    private final int TASK_NUMBER;
```
is TASK_NUMBER a constant value? Use camelCase instead of all uppercase if it's not a constant. To find what exactly are considered constants, you might want have a look on this page https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Casielim commented on src/main/java/ron/command/Parser.java:
```diff
@@ -0,0 +1,67 @@
+        String[] tokens = command.trim().split(" ", 2);
+        String action = tokens[0];
+
+        switch (action) {
```
IIRC,  there is no need to have indentation for case clauses.

Casielim commented on src/test/java/ron/storage/StorageTest.java:
```diff
@@ -0,0 +1,90 @@
+    public void testSaveAndLoadTasks() throws RonException {
+        ArrayList<Task> tasksToSave = new ArrayList<>();
+        tasksToSave.add(new Todo("clubbbbbbbbbbb"));
+        tasksToSave.add(new Deadline(" 2 leetcodeeee questions", LocalDateTime.parse("2025-01-31 18:00", DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))));
```
Try to keep line length shorter than 110 characters. You might want to use line wrapping.

Casielim commented on src/test/java/ron/storage/StorageTest.java:
```diff
@@ -0,0 +1,90 @@
+        ArrayList<Task> tasksToSave = new ArrayList<>();
+        tasksToSave.add(new Todo("clubbbbbbbbbbb"));
+        tasksToSave.add(new Deadline(" 2 leetcodeeee questions", LocalDateTime.parse("2025-01-31 18:00", DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))));
+        tasksToSave.add(new Event("iex meetinggg", LocalDateTime.parse("2025-02-01 09:00", DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
```
Try to keep line length shorter than 110 characters to ensure better readability.

Casielim reviewed (commented):
Looks good to me!
Some minor changes that do not affect functionality 😄

salads-source committed:
Add additional unit test for event task

salads-source committed:
Edit javadocs for ui methods and change styling for gui

salads-source committed:
Modify gradle build file for jar file packaging

salads-source committed:
Add assertion checks to TaskList and Parser

TaskList and Parser currently do not validate certain assumptions
before performing operations, which may lead to runtime errors.

This commit adds assertions to:
- TaskList: Ensure task index is within valid bounds before marking/unmarking.
- Parser: Validate that input commands are non-null and contain required arguments.

These changes help detect potential issues early and improve the
robustness of the codebase.

salads-source committed:
Refactor code to improve readability and maintainability

The existing code contains deeply nested logic, magic numbers, and
direct print statements, making it harder to read and modify.
Additionally, TaskList and Parser contain redundant code that can
be simplified.

This commit improves code quality by:
- Extracting validation logic in TaskList to reduce code duplication.
- Refactoring Parser to use Command classes for better structure.
- Replacing magic strings in Storage with named constants.
- Updating Ui methods to return Strings instead of printing directly
  (to support JavaFX UI).
- Simplifying the flip() method in DialogBox for better readability.

These changes follow best practices from the Code Quality module,
making the codebase cleaner and easier to maintain.

salads-source committed:
Merge pull request #1 from salads-source/branch-A-Assertions

Add assertion checks to TaskList and Parser

salads-source committed:
Merge branch 'master' into branch-A-CodeQuality

salads-source committed:
Merge pull request #2 from salads-source/branch-A-CodeQuality

Refactor code to improve readability and maintainability

salads-source committed:
Set up GitHub Actions CI for automated testing

This commit adds a GitHub Actions workflow to automate the build and
test process for the project.

Key features:
- Runs on Linux, macOS, and Windows using a matrix strategy.
- Checks out the repository and validates the Gradle wrapper.
- Sets up Java 17 with JavaFX.
- Builds and runs tests using Gradle.
- Performs input/output redirection tests on all platforms.

This ensures that new changes are automatically verified for correctness
before merging.

salads-source committed:
Add help command to provide in-app guidance

The application currently lacks a way for users to view available commands,
which may make it harder for new users to understand how to interact with the app.

This commit introduces:
- HelpCommand: A new command that displays a list of all available commands.
- showHelp() in Ui: A method that formats and prints the list of commands.
- Updates Parser to recognisethe help command.

salads-source committed:
Merge branch 'branch-C-Help' into 'master'

salads-source committed:
Modify workflow yml file

salads-source committed:
Modify styling for GUI for better appeal

salads-source committed:
Add Ui.png to docs folder

salads-source committed:
Add README for github pages

salads-source committed:
Fix styling issues with GUI window

salads-source committed:
Add references and credit for ChatGPT for where and how it was used

salads-source committed:
Generate jar file for version release
